xonsh/events.py

```diff
@@ -0,0 +1,113 @@
+"""
+Events for xonsh.
+
+In all likelihood, you want builtins.__xonsh_events__
+
+The best way to "declare" an event is something like::
+
+    __xonsh_events__.on_spam.__doc__ = "Comes with eggs"
+"""
+
+
+class Event(set):
```

Comment:
Well, it's not randomized, it's hash order. So each session will run them in the same order, and a given installation and configuration may always run them in the same order.

Xontribs will load in any order. The real question is, how does amalgamate order modules?